### PR TITLE
Add `--ignore-if-missing` flag to `github retarget-pr` command

### LINKED ISSUE #877: Make `github retarget-pr` accept `--ignore-if-missing` flag
Mostly for the use in scripts. If there's no PR for the given branch (or the branch doesn't even exist in the remote repo), let's still exit with status code 0.

Prereq for #875.

### PR DESCRIPTION
**Pull request by amalota**
TODO:
- [x] implement the new flag
- [x] add test case
- [x] update docs
- [x] update command completions

### CODE REVIEW

File: git_machete/client.py [hunk 1]
```diff
@@ -2168,15 +2168,15 @@ def __get_added_remote_name_or_none(self, github_domain: str, remote_url: str) -
...
         domain = self.__derive_github_domain()
         remote_org_repo = self.__derive_remote_and_github_org_and_repo(domain=domain, branch_used_for_tracking_data=head)
         github_client = GitHubClient(domain=domain, organization=remote_org_repo.organization, repository=remote_org_repo.repository)
 
         debug(f'organization is {remote_org_repo.organization}, repository is {remote_org_repo.repository}')
 
         pr: Optional[GitHubPullRequest] = github_client.derive_pull_request_by_head(head)
-        if not pr:
+        if not pr and not ignore_if_missing:
             raise MacheteException(f'No PR is opened in {bold(remote_org_repo.organization)}/{bold(remote_org_repo.organization)} '
```
Comment (PawelLipski) [inline git_machete/client.py]: Even if `ignore_if_missing == True`, the execution must **not** continue when the PR is missing

Comment (amalota) [reply git_machete/client.py]: @amalota By not continuing you mean raising the exception or just quietly terminating execution?

Comment (PawelLipski) [reply git_machete/client.py]: Quietly terminating execution! as raising the exception is the behavior that happens without `--ignore-if-missing`

File: git_machete/client.py [hunk 2]
```diff
@@ -2168,17 +2168,16 @@ def __get_added_remote_name_or_none(self, github_domain: str, remote_url: str) -
...
-    def retarget_github_pr(self, head: LocalBranchShortName) -> None:
+    def retarget_github_pr(self, head: LocalBranchShortName, ignore_if_missing: bool) -> None:
         domain = self.__derive_github_domain()
         remote_org_repo = self.__derive_remote_and_github_org_and_repo(domain=domain, branch_used_for_tracking_data=head)
         github_client = GitHubClient(domain=domain, organization=remote_org_repo.organization, repository=remote_org_repo.repository)
 
         debug(f'organization is {remote_org_repo.organization}, repository is {remote_org_repo.repository}')
 
-        pr: Optional[GitHubPullRequest] = github_client.derive_pull_request_by_head(head)
+        pr: Optional[GitHubPullRequest] = github_client.derive_pull_request_by_head(head, ignore_if_missing)
```
Comment (PawelLipski) [inline git_machete/client.py]: Hmm I'd rather **not** pollute `GitHubClient` with details like `ignore_if_missing`... let's throw an exception in `GitHubClient.derive_pull_request_by_head`, and handle it here in `retarget_github_pr` (possibly ignoring it, if `ignore_if_missing` is True)

**Review by PawelLipski — APPROVED** [review]
:benissimo:

**Pull request merged**